py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,587 @@
+        Available placeholders:
+            icon, coverage
+        (default '{icon}: {coverage}%')
+    format_humidity: Formatting for humidity (percentage)
+        Available placeholders:
+            icon, humid
+        (default '{icon}: {humid}%')
+    format_pressure: Formatting for atmospheric pressure
+        Available placeholders:
+            icon, press, sea_level
+        (default '{icon}: {press} hPa')
+    format_rain: Formatting for rain volume over the past 3 hours
+        Available placeholders:
+            icon, mm, cm, m, in, ft, yrd
+        (default '{icon}: {in} inches')
+    format_snow: Formatting for snow volume over the past 3 hours
+        Available placeholders:
+            icon, mm, cm, m, in, ft, yrd
+        (default '{icon}: {in} inches')
+    format_sunrise: Formatting for sunrise time.
+        Note that this format accepts strftime/strptime placeholders to populate
+        the output with the time information.
+        Available placeholders:
+            icon
+        (default '{icon}: %X')
+    format_sunset: Formatting for sunset time.
+        Note that this format accepts strftime/strptime placeholders to populate
+        the output with the time information.
+        Available placeholders:
+            icon
+        (default '{icon}: %X')
+    format_temp: Formatting for temperature
+        Available placeholders:
+            icon, c, c_min, c_max, f, f_min, f_max, k, k_min, k_max
+        (default '{icon}: {f}°')
+    format_wind: Formatting for wind degree and speed
+        Available placeholders:
+            icon, deg, msec_speed, kmh_speed, fsec_speed, mph_speed,
+            msec_gust, kmh_gust, fsec_gust, mph_gust
+        (default '{icon}: {mph_speed} mph')
+    icons: A dictionary relating weather code to icon.
+        See https://openweathermap.org/weather-conditions for a complete list
+        of supported icons. This will fall-back to the listed icon if there is
+        no specific icon present.
+        There are multiple ways to specify individual icons based on the id:
+            - Use the key 'i601' to reference the condition with id = 601
+              (snow)
+            - Use the key 'i230_i232' to reference a span of conditions
+              inclusive, in this case conditions (230, 231, 232) (thunderstorm
+              with drizzle)
+        Also, you can specify the icons for the various formatting sections
+        below. For example, to specify the icon for 'format_pressure', use
+        'pressure'. A few formatting sections will take from the defaults or be
+        dynamic, including
+            - clouds
+            - rain
+            - wind
+            - sunrise
+            - sunset
+        These may be specified regardless.
+        (default None)
+    lang: An ISO 639-1 code for your language (two letters)
+        (default 'en')
+    req_timeout: The timeout in seconds for contacting the IP API.
```

Comment:
basically yes we should be moving towards 'universal' options if we can, for example move all modules to use `py3.request()` then it can just look at the modules config to get the timeout.  I'd go with `request_timeout` as

* it is the most popular

* the `request_` prefix shows what it affects so clarity

 * you are in favour

We could also allow it to be set in the `py3status {}` config section so universal.  Getting things into a state where we can do that would be good imho